pkg/operator/controller/canary/route_test.go

```diff
@@ -19,7 +19,11 @@ func TestDesiredCanaryRoute(t *testing.T) {
 		Name: "test",
 	}
 	service := desiredCanaryService(daemonsetRef)
-	route := desiredCanaryRoute(service)
+	route, err := desiredCanaryRoute(service)
+
+	if err != nil {
+		t.Errorf("desiredCanaryService returned an error: %v", err)
```

Comment:
No sense in continuing with the unit test if we don't have a service, right?  
```suggestion
		t.Fatalf("desiredCanaryService returned an error: %v", err)
```